Add es localization

alosarjos opened the pull request:
Spanish (Spain) localization. 

PD: Thanks for the paru project. Always nice to see more Rusty stuff

Morganamilo commented:
I'm not that familiar with Spanish or localisation in general. Pacman has its spanish translation as just `es.po` Is there a reason you picked the more specific es_ES? Is there a reason this couldn't just be es? As a pacman wrapper I try to follow what pacman does so it lines up as best as possible.

brunequerequeteque commented:
Not from Spain Spanish speaker here. For most cases is not different than different English localisations. I am sure that if there is some region specific wording it may be changed to a more regional independent synonim and therefore there can be just one Spanish localisation.

Morganamilo commented:
That would be my preference as pacman just ships a `es` and not `es_ES`  https://gitlab.archlinux.org/pacman/pacman/-/tree/master/src/pacman/po

brunequerequeteque commented:
I've just quickly read the contribution and I think it can perfectly be a whole es localisation. Great job alosarjos.

Morganamilo commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+"Report-Msgid-Bugs-To: https://github.com/Morganamilo/paru\n"
+"POT-Creation-Date: 2021-06-28 03:24+0000\n"
+"PO-Revision-Date: 2021-06-28 18:02+0200\n"
+"Language: es_ES\n"
```
This should be changed along with the commit message.

alosarjos replied:
My bad, should be fixed

alosarjos commented:
Changed filename to es.po, no problem on my end if that's fine with you

Morganamilo commented:
Should all be good now. Thanks a lot.

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/clean.rs:31
+msgid "Do you want to remove ALL AUR packages from cache?"
+msgstr "¿Quieres eliminar TODOS los paquetes de AUR de la caché?"
```
```suggestion
msgstr "¿Quiere eliminar TODOS los paquetes del AUR del caché?"
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/clean.rs:33
+msgid "Do you want to remove all other AUR packages from cache?"
+msgstr "¿Quieres eliminar todos los demás paquetes de AUR de la caché?"
```
```suggestion
msgstr "¿Quiere eliminar todos los demás paquetes del AUR del caché?"
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/clean.rs:48
+msgid "Do you want to remove all saved diffs?"
+msgstr "¿Quieres eliminar todos los diffs guardados?"
```
```suggestion
msgstr "¿Quiere eliminar todos los diffs guardados?"
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/clean.rs:93
+msgid "can't open clone dir: {}"
+msgstr "no se puede abrir el directorio clonado: {}"
```
```suggestion
msgstr "no pudo abrir el directorio clonado: {}"
```

I like how you translated "dir" to "directorio" here so I think consistency is good.

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/command_line.rs:127
+msgid "option {} expects a value"
+msgstr "opción {} espera un valor"
```
```suggestion
msgstr "la opción {} requiere un valor"
```

The exact verb to use is largely irrelevant, but I always see "la opción..." in other translations

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/command_line.rs:158 src/command_line.rs:327
+msgid "option {} does not allow a value"
+msgstr "opción {} no admite un valor"
```
```suggestion
msgstr "la opción {} no admite un valor"
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/completion.rs:27
+msgid "failed to open cache file '{}'"
+msgstr "error al abrir desde caché el fichero '{}'"
```
```suggestion
msgstr "error al abrir el fichero '{}' desde el caché"
```

Morganamilo replied:
I use failed because errors are prefixed with `error:`. So when printed it's `error: failed to open cache file`. This means it's going to say error twice. Would be nice if that could be avoided.

Morganamilo replied:
Yep `error: could not open...` sounds good to me. As long as it sounds good in Spanish too.

em-may replied:
Fine by me as well, sounds good in Spanish too.

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/completion.rs:65
+msgid "could not update aur cache"
+msgstr "no se puedo actualizar la caché de aur"
```
```suggestion
msgstr "no se pudo actualizar la caché de aur"
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/config.rs:550
+msgid "only one operation may be used at a time"
+msgstr "solo se puede usar una operación a la vez"
```
```suggestion
msgstr "solo se puede realizar una operación a la vez"
```

Personal semantic preference, just a suggestion.

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/config.rs:740 src/config.rs:873
+msgid "value can not be empty for key '{}'"
+msgstr "la clave '{}' no admite un valor vacio"
```
```suggestion
msgstr "la clave '{}' no admite un valor vacío"
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/config.rs:756
+msgid "key '{}' does not belong to a section"
+msgstr "clave '{}' no pertenece a sección"
```
```suggestion
msgstr "la clave '{}' no pertenece a una sección"
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/config.rs:769 src/config.rs:824 src/config.rs:829
+msgid "key can not be empty"
+msgstr "clave no puede estar vacia"
```
```suggestion
msgstr "la clave no puede estar vacia"
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/config.rs:791
+msgid "error: unknown option '{}' in section [bin]"
+msgstr "error: opción desconocida '{}' es sección [bin]"
```
```suggestion
msgstr "error: opción desconocida '{}' en sección [bin]"
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/config.rs:891
+msgid "error: unknown option '{}' in section [options]"
+msgstr "error: opción desconocida '{}' es sección [options]"
```
```suggestion
msgstr "error: opción desconocida '{}' en sección [options]"
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+#: src/config.rs:961 src/info.rs:56 src/info.rs:95 src/install.rs:1755
+#: src/install.rs:1804
+msgid "Repository"
+msgstr "Repositorio"
```
```suggestion
msgstr "Depósito"
```

alosarjos replied:
At least on my circles, people tend to use more Repositorio than Depósito

em-may replied:
This is my preference but I don't have a super strong opinion.

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/devel.rs:129 src/download.rs:179
+msgid "Querying AUR..."
+msgstr "Consultando AUR..."
```
```suggestion
msgstr "Buscando el AUR..."
```

alosarjos replied:
"Buscando el AUR" is more like "Search the AUR". If using the word "Buscando" I would rather use it as "Buscando en el AUR"

em-may replied:
That's acceptable to me.

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/devel.rs:142 src/devel.rs:166 src/install.rs:694
+msgid "failed to parse srcinfo for '{}'"
+msgstr "error al parsear srcinfo para '{}'"
```
```suggestion
msgstr "error al leer srcinfo para '{}'"
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/devel.rs:382
+msgid "package does not have an update"
+msgstr "paquete no tiene actualización"
```
```suggestion
msgstr "el paquete no tiene actualización"
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/download.rs:83
+msgid "packages not in the AUR: "
+msgstr "paquetes que no están en AUR: "
```
```suggestion
msgstr "paquetes que no están en el AUR: "
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/download.rs:210
+msgid "can not get repo packages: asp is not installed"
+msgstr "no se pueden obtener paquetes desde repositorio: asp no esta instalado"
```
```suggestion
msgstr "no se pudo obtener paquetes desde repositorio: asp no esta instalado"
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/download.rs:520
+msgid "{} is not installed: can not get repo packages"
+msgstr "{} no esta instalado: no se pueden obtener paquetes desde repositorio"
```
```suggestion
msgstr "{} no está instalado: no se pudo obtener paquetes desde el depósito"
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/exec.rs:86
+msgid "Pacman is currently in use, please wait..."
+msgstr "Pacman se esta ejecutando, por favor espera..."
```
```suggestion
msgstr "Ya se está ejecutando Pacman, por favor espera..."
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/fmt.rs:6 src/info.rs:146 src/search.rs:176
+msgid "None"
+msgstr "Nada"
```
```suggestion
msgstr "Ninguno"
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/info.rs:30 src/query.rs:51
+msgid "package '{}' was not found"
+msgstr "no se pudo encontrar paquete '{}'"
```
```suggestion
msgstr "no se pudo encontrar el paquete '{}'"
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/info.rs:67 src/info.rs:114
+msgid "Maintainer"
+msgstr "Mantenedor"
```
```suggestion
msgstr "Encargado"
```

This is the official Pacman translation btw.

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/install.rs:261 src/install.rs:1462
+msgid "{}-{} is up to date -- skipping build"
+msgstr "{}-{} esta actualizado -- saltando construcción"
```
```suggestion
msgstr "{}-{} está actualizado -- saltando construcción"
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/install.rs:334
+msgid "could not get news"
+msgstr "no se pudieron obtener noticias"
```
```suggestion
msgstr "no se pudo obtener noticias"
```

Third person singular because a tool is doing the work, not a group of tools/persons/etc.

Morganamilo replied:
This and others. Is the conversation elsewhere?

alosarjos replied:
https://github.com/Morganamilo/paru/pull/435#discussion_r660050859

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/install.rs:339 src/install.rs:521 src/install.rs:716 src/install.rs:848
+msgid "Proceed with installation?"
+msgstr "¿Proceder con la instalación?"
```
```suggestion
msgstr "¿Continuar con la instalación?"
```

Personal preference but this appears elsewhere fyi.

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/install.rs:356
+msgid "no targets specified (use -h for help)"
+msgstr "no se han seleccionado objetivos (usa -h para ayuda)"
```
```suggestion
msgstr "ningún objectivo seleccionado (usa -h para ayuda)"
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+#: src/install.rs:434 src/install.rs:461 src/search.rs:324 src/search.rs:360
+#: src/lib.rs:227
+msgid " there is nothing to do"
+msgstr " nada por hacer"
```
```suggestion
msgstr " no hay nada que hacer"
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/install.rs:518
+msgid "Proceed to review?"
+msgstr "¿Proceder a revisión?"
```
```suggestion
msgstr "¿Editar ahora?"
```

This is a pretty common translation from other helpers/pkgs.

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/install.rs:559
+msgid "The following packages are not compatible with your architecture:"
+msgstr "Los siguientes paquetes no son compatibles con tu arquitectura:"
```
```suggestion
msgstr "Los siguientes paquetes no son compatibles con su arquitectura:"
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/install.rs:662
+msgid "packages failed to build: {}"
+msgstr "paquetes que no se pudieron construir: {}"
```
```suggestion
msgstr "no se pudo construir los siguentes paquetes: {}"
```

alosarjos replied:
I'm pretty sure it should be "pudieron" since you are refereing to multiple things

em-may replied:
Ah, I suppose since it's reflexive that's true.

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/install.rs:698
+msgid "could not find .SRINFO for '{}'"
+msgstr "no se puedo encontrar .SRINFO para '{}'"
```
```suggestion
msgstr "no se pudo encontrar .SRINFO para '{}'"
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/install.rs:874
+msgid "file manager did not execute successfully"
+msgstr "gestor de archivos no pudo ejecutar correctamente"
```
```suggestion
msgstr "el gestor de archivos no pudo ejecutar correctamente"
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/install.rs:975
+msgid "could not find all required packages:"
+msgstr "no se pudieron encontrar todos los paquetes necesarios:"
```
```suggestion
msgstr "no se pudo encontrar todos los paquetes necesarios:"
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/install.rs:1411
+msgid "{}: parsing pkg list..."
+msgstr "{}: parseando lista pkg..."
```
```suggestion
msgstr "{}: leyendo lista pkg..."
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/install.rs:1670
+msgid "can't find package name in packagelist: {}"
+msgstr "no se puede encontrar el nombre del paquete en lista de paquetes: {}"
```
```suggestion
msgstr "no se pudo encontrar el nombre del paquete en lista de paquetes: {}"
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+msgstr ""
+"\n"
+"\n"
+"Introduce una elección (default=all): "
```
```suggestion
"Elegir una opción (default=all): "
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/install.rs:1965
+msgid "{}-{} is up to date -- skipping install"
+msgstr "{}-{} esta actualizado -- saltando instalación"
```
```suggestion
msgstr "{}-{} está actualizado -- saltando instalación"
```

alosarjos commented:
Updated the Yes/No responses to spanish. Thought they wouldn't work and would make people insert an invalid value when asked

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/keys.rs:47
+msgid "keys need to be imported:)"
+msgstr "claves necesitan importarse :)"
```
```suggestion
msgstr "se necesitan importar claves :)"
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/repo.rs:164
+msgid "failed to get current exe"
+msgstr "error al obtener exe actual"
```
```suggestion
msgstr "error al obtener binario actual"
```

alosarjos replied:
This exe is referering to a binary or an executable? executable would be better translated as "ejecutable" rather than "binario"

Morganamilo replied:
binary and executable are the same thing really.

em-may replied:
Yes, executable would be translated as ejecutable imo as well, but since it says exe I went with binary. Either way, I think exe should be replaced because it requires the reader to know the English spelling of executable, which defeats the purpose of translations.

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/search.rs:321
+msgid "Packages to install (eg: 1 2 3, 1-3):"
+msgstr "Paquetes a instalar (ej: 1 2 3, 1-3):"
```
```suggestion
msgstr "Paquetes para instalar (ej: 1 2 3, 1-3):"
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/stats.rs:87
+msgid "Aur packages: {}"
+msgstr "Paquetes de AUR: {}"
```
```suggestion
msgstr "Paquetes del AUR: {}"
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/util.rs:299
+msgid "Enter a number (default=1): "
+msgstr "Introduce un número (Por defecto = 1): "
```
```suggestion
msgstr "Introduzca un número (Por omisión = 1): "
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/util.rs:326
+msgid "invalid value: {:n} is not between 1 and {:max}"
+msgstr "valor inválido: {:n} no esta entre 1 y {:max}"
```
```suggestion
msgstr "valor inválido: {:n} no está entre 1 y {:max}"
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/clean.rs:64
+msgid "can't open diff dir: {}"
+msgstr "no se puede abrir el dir de diff: {}"
```
```suggestion
msgstr "no se pudo abrir el dir de diff: {}"
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/config.rs:638
+msgid "can not find local repo '{}' in pacman.conf"
+msgstr "no se puede encontrar el repositorio local '{}' en pacman.conf"
```
```suggestion
msgstr "no se pudo encontrar el repositorio local '{}' en pacman.conf"
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/install.rs:138
+msgid "can't build package as root"
+msgstr "no se puede construir paquete como root"
```
```suggestion
msgstr "no se pudo construir paquete como root"
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/install.rs:488
+msgid "can't install AUR package as root"
+msgstr "no se puede instalar un paquete de AUR como root"
```
```suggestion
msgstr "no se permite instalar un paquete de AUR como root"
```

em-may commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/lib.rs:153
+msgid "can not use chroot builds: devtools is not installed"
+msgstr "no se puede construir en chroot: devtools no esta instalado"
```
```suggestion
msgstr "no se pudo construir en chroot: devtools no esta instalado"
```

em-may reviewed (changes requested):
Some quick notes for my review:
* Usted form should always be used
* "caché" is masculine (even in Spanish)
* AUR is "el AUR" IMO, since it's "el depósito..." or "el almacén..."
* Be consistent with "no se pudo" (IMO don't use "no se puede" in spanish)
* Be consistent with translating "dir" to "directorio" or just "dir" (not my favorite, but consistency is important either way)

alosarjos commented:
> * "caché" is masculine (even in Spanish)

I'm not sure about this, since caché  takes the genre of the element it refers to, which usually (And I think in this case) is memory, which is femenine in spanish, so it's called "La memoria caché" or "La caché".

> * AUR is "el AUR" IMO, since it's "el depósito..." or "el almacén..."

Since AUR is something unique I think it's better without el, in the same way in English it's "from AUR" and not "from the AUR"

> * Be consistent with "no se pudo" (IMO don't use "no se puede" in spanish)

I used puede and pudo based on the time (can/can't or could(couldn't) that comes from the English version

> * Be consistent with translating "dir" to "directorio" or just "dir" (not my favorite, but consistency is important either way)


> * Usted form should always be used
> * Be consistent with translating "dir" to "directorio" or just "dir" (not my favorite, but consistency is important either way)

I agree with these two. Will apply them as soon as I get back home

em-may commented:
> I'm not sure about this, since caché takes the genre of the element it refers to, which usually (And I think in this case) is memory, which is femenine in spanish, so it's called "La memoria caché" or "La caché".

Fair enough.

> Since AUR is something unique I think it's better without el, in the same way in English it's "from AUR" and not "from the AUR"

In English you say "from the AUR".

> I used puede and pudo based on the time (can/can't or could(couldn't) that comes from the English version

Anectodally, I've seen Pacman, other AUR helpers, and other projects translate phrases like "can't open directory" or "couldn't find package" as "no se pudo...", while phrases which indicate things that worked, are ongoing, or will happen use simple present or present progressive forms (i.e. "no se ha...", and "va a estar..."). This is somewhat of a personal/stylistic choice, but I'll cite an example from running Pacman when the db is locked (my locale is es_ES.UTF-8):

```
$ sudo pacman -S asdf
error: **no se pudo** iniciar la operación (**no se pudo** bloquear la base de datos)
error: **no se pudo** bloquear la base de datos: El fichero **ya existe**
  **si está seguro** de que **no se está ejecutando**
  un administrador de paquetes, **puede quitar** /var/lib/pacman/db.lck
```

Emphasis mine. Of course, I see the argument for a direct translation, however I prefer translating to match the grammatical styles that I've been most exposed to in whatever the target language is (in this case Spanish), as noted above.

Morganamilo commented:
Yep following pacman should come above anything else.

alosarjos commented:
I updated the translation based on the provided feedback. Replaced the strings starting with failed by equivalent of "could not".  No problem using that in spanish.

Morganamilo commented on po/es.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/completion.rs:27
+msgid "failed to open cache file '{}'"
+msgstr "error al abrir el fichero '{}' desde la caché"
```
There's still plenty of failed => errorm

alosarjos commented:
All the "failed" should be fixed

Morganamilo commented:
What about @em-may's suggestions?

alosarjos commented:
There should be most if not all of them (Some of those were discussed and left them with what both agree). I hope I didn't miss anything.

Morganamilo commented:
Okay then looks good. Thanks.
